raster-foundry/raster-foundry
# Enhance and fix datasources list view

## Overview
Fix datasource loading spinner
Rewrite fetching logic to make user of search parameter
Enhance pagination and searchbox components to pull from state parameters when initialized.
Add search param to a couple endpoints

### Checklist

- ~Styleguide updated, if necessary~
- ~Symlinks from new migrations present or corrected for any new migrations~
- ~Any content changes are properly templated using `BUILDCONFIG.APP_NAME`~
- ~Any new SQL strings have tests~

### Demo
![image](https://user-images.githubusercontent.com/4392704/44235966-660c5880-a179-11e8-80f6-4f01e5ba314a.png)

## Testing Instructions

 * Verify that the loading message goes away once datasources have been loaded
* verify that searching works
* verify that pagination works
* verify that if you reload the page, search and page are preserved
* verify that clearing the search works as expected
* verify that search counts are accurate, and the transcluded search term message next to the search count makes sense
![image](https://user-images.githubusercontent.com/4392704/44236615-a2d94f00-a17b-11e8-8d37-8d51df8b51e2.png)

### Testing round 2
For each view listed, do the following:
* Verify that the item type described in all error / placeholder / count messages are correct
* Verify that the error message shows when a request fails. Simulate this by using dev tools to set the page in offline mode. Also verify that the pagination dissapears
![image](https://user-images.githubusercontent.com/4392704/44537543-18d63c80-a6cd-11e8-8273-a15b8a934df3.png)
* Verify that searching functions correctly in every view where it appears
* Verify that searching a string with no matches has the correct error message
* Verify that pagination functions correctly with and without a search defined
* Verify that the view preserves pagination and search on reload
* Verify that pagination and search are correctly represented in the URL
* Verify that counts are not visible when there are no items in the list (either from a search or when there are none to begin with)

#### Views to test
- [x] datasource delete modal (simple text fix, no pagination here)
- [x] project select modal
- [x] admin/organization/projects
- [x] admin/organization/raster
- [x] admin/organization/teams
- [x] admin/organization/users
- [x] admin/platform/organizations 
- [x] admin/platform/projects
- [x] admin/platform/users
- [x] admin/team/projects
- [x] admin/team/rasters
- [x] admin/team/users
- [x] imports/datasources/list
- [x] imports/vector
- [x] lab/browse/analyses
- [x] lab/browse/templates
- [x] projects/list - also verify that the owned by me / shared with me filter works
- [x] user/projects
- [x] user/rasters

### Notes round 2
Not every page was fully updated - part way through I decided not to continue updating views that aren't actually used or linked to. They've all been updated to the new style, but don't necessarily have the correct text and stuff.


Closes https://github.com/raster-foundry/raster-foundry/issues/3123
Closes #3868 
Closes #3916
